gax/operations_stub.h

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAPIC_GENERATOR_CPP_GAX_OPERATIONS_STUB_H_
+#define GAPIC_GENERATOR_CPP_GAX_OPERATIONS_STUB_H_
+
+#include "google/longrunning/operations.pb.h"
+#include "gax/call_context.h"
+#include "gax/status.h"
+
+namespace google {
+namespace gax {
+
+class OperationsStub {
+ public:
+  virtual ~OperationsStub() = 0;
+
+  virtual gax::Status GetOperation(
+      gax::CallContext& context,
+      google::longrunning::GetOperationRequest const& request,
+      google::longrunning::Operation* response);
+
+  virtual gax::Status DeleteOperation(
+      gax::CallContext& context,
+      google::longrunning::DeleteOperationRequest const& request,
+      google::protobuf::Empty* response);
+
+  virtual gax::Status CancelOperation(
+      gax::CallContext& context,
+      google::longrunning::CancelOperationRequest const& request,
+      google::protobuf::Empty* response);
+
+  static constexpr MethodInfo get_operation_info = {
```

Comment:
If I understand it right, it is somewhat strange mix of `CallContext` and `MethodInfo` on the surface of same class. `OperationsStub` methods operate on `gax::CallContext`, which encapsulates `MethodInfo`. I assume that these static `constexprs` are needed to create the proper `CallContext`, for example (from tests):
```C++
gax::CallContext getCtx(gax::OperationsStub::get_operation_info);
```
How user-facing is this class? What is the chance, that anybody who uses, lets say `OperationsStub::GetOperation()` will ever need to call it with a context different from the default one (like the one created in the code snippet above). Would it make sense to introduce the concept of "default" call context? So people can just call `GetOperation(request, &response)` (seems like what most people would want)?

Reply:
`CallContext` has an accessor for its method info so that user-written stub decorators, some of which will probably be autogenerated, can view streaming/unary or idempotency. I'm going to move these method infos to the new OperationsClient class, which is more in line with what the GAPIC clients are going to have.